File: main (2).cs
using System;
using System.Collections.Generic;

class Program
{
    static void Main()
    {
        var rnd = new Random();
        char opcion;

        do
        {
            var numeros = new List<int>();
            while (numeros.Count < 4)
            {
                int num = rnd.Next(1, 100);
                if (!numeros.Contains(num))
                {
                    numeros.Add(num);
                }
            }

            Console.Write("Lista original: ");
            numeros.ForEach(n => Console.Write(n + " "));
            Console.WriteLine();


            bool swapped = true;
            while (swapped)
            {
                swapped = false;
                for (int i = 0; i < numeros.Count - 1; i++)
                {
                    if (numeros[i] > numeros[i + 1])
                    {
                       
                        int temp = numeros[i];
                        numeros[i] = numeros[i + 1];
                        numeros[i + 1] = temp;
                        swapped = true;
                    }
                }
            }

            Console.Write("Lista ordenada: ");
            numeros.ForEach(n => Console.Write(n + " "));
            Console.WriteLine();

            
            Console.Write("¿Quieres generar y ordenar otra lista? (S/N): ");
            opcion = Console.ReadLine()[0];
        }
        while (char.ToLower(opcion) == 's');
    }
}

File: main (2).js
function generarListaAleatoria() {
    let numeros = [];
    while (numeros.length < 4) {
        let num = Math.floor(Math.random() * 99) + 1;
        if (!numeros.includes(num)) {
            numeros.push(num);
        }
    }
    return numeros;
}

function bubbleSort(lista) {
    let swapped;
    do {
        swapped = false;
        for (let i = 0; i < lista.length - 1; i++) {
            if (lista[i] > lista[i + 1]) {
                [lista[i], lista[i + 1]] = [lista[i + 1], lista[i]];
                swapped = true;
            }
        }
    } while (swapped);
    return lista;
}

function iniciar() {
    let continuar = true;
    while (continuar) {
        let lista = generarListaAleatoria();
        console.log("Lista original:", lista);
        let ordenada = bubbleSort([...lista]);
        console.log("Lista ordenada:", ordenada);

        let respuesta = prompt("¿Quieres generar y ordenar otra lista? (S/N): ");
        if (!respuesta || respuesta.toLowerCase() !== "s") {
            continuar = false;
        }
    }
}

iniciar();


